### ISSUE TITLE
Production bot not automatically collecting finished items

### ISSUE THREAD

**Issue by andrewcharlwood** [Th3C0D3R/FoBJS_Release#3]
Which Server are you playing on (de/en/beta(zz)/...) - en
What OS you run - Windows 10 Pro
Are there any errors on the screen - No
Expected behavior - After starting production bot, the bot should automatically harvest and restart productions once completed
Actual behavior - The production bot works perfectly, however it seems to be unable to tell if items are completed or not until the screen is refreshed.

For example, if production bot is running and the gui shows my buildings are idle, it will automatically start producing the correct items.

Similarly, once the items are showing "finished" in the gui, it correctly collects & restarts production.

However, if the gui is showing "Producing", it will maintain this status until either click  "update lists", or running another command that refreshes the page (e.g. aiding others, visiting taverns, collecting silver etc)

Possible solution: (Although my knowledge of coding is extremely limited as i'm just learning so take this with a pinch of salt....)

I assume this problem has arisen since removing the production bot timer.

Could it be possible to set a page reload function to trigger based on the production time selected (e.g. 5 mins, 15 mins etc) + 30 seconds (to ensure the products are finished on reload)?

Or set a refresh script to run every 2:30?

**Comment by Th3C0D3R** [comment]
Yeah, this is under development because I encountered the issue with high CPU usage if the productiontimer is updated every second on the page. 
I thought about moving the timer to the frontend (was backend before) but have to test it (which unfortunately I cannt do before Monday)

The idea about a refresh after the production + 30 and the refresh after some time is good, but if the user closes the app and open it again and start the bot (because he closed the app accidentally) it would take more time. also I have to make some requests to the servers to get the state of the production

I have to look whats the best to do (because I liked the fact that it shows on the page how long its going to take) but it can take some time to implement to right (doing this in my freetime)

But thanks for the report. every Issue/Bug is a welcomed one

**Comment by andrewcharlwood** [comment]
Of course, don't take my reports as a complaint in any shape or form, just trying to help!

I understand this is a hobby so I hope i'm not coming across impatient.

>The idea about a refresh after the production + 30 and the refresh after some time is good, but if the user closes the app and open it again and start the bot (because he closed the app accidentally) it would take more time. also I have to make some requests to the servers to get the state of the production



Assuming that all productions are synced up together, and one would expect they would as from what i understand that's the most efficient way of increasing production. Could you set a global variable which is the ETA of completion of all productions? 

Instead of having multiple timers for each production line, have a time stamp which when reached sends a request to the server to confirm status?

I literally started playing the game like, 3 days ago though, so, forgive my ignorance if this wouldn't work.

**Comment by Th3C0D3R** [comment]
Na its all right ;) I will see if there is something I can do. A global ETA is a good idea. Have to look how to implement it. 

Have an idea right now in my head which, if working is a perfect "fix" for that

**Comment by Th3C0D3R** [comment]
@andrewcharlwood can you report to me if the bot is working?
Because for me its not. And if its also not working for you than I have to work on it again

**Comment by andrewcharlwood** [comment]
Hey,

It's been a bit patchy in all honesty. It was working fine when I had a large number of blacksmiths, I then recently upgraded to Butchers and found it was really hit and miss whenever I had one blacksmith being built (I think it would report that because one was idle, that they were all idle).

I also noticed if I was logged in on my phone, it would absolutely have a hissy fit and refresh repeatedly (I imagine as it'd have timed out), easily solved by logging out of the phone.

I've restarted it running in the last 30 minutes since all my butchers have completed and it seems to be working. However last night it seemed to be working up until the 1 hour mark I think (based on the increase in production)

I'll leave it running for a further hour and update if it stops working at any time.

**Comment by andrewcharlwood** [comment]
Forgot to mention, after new buildings had been completed I had to clear userdata to recognise they were completed and then it started working again.

**Comment by Th3C0D3R** [comment]
Okay thank you 🤔 have to work on it again

**Comment by andrewcharlwood** [comment]
Does it work for you if after you start the bot running, you press function - refresh?

**Comment by andrewcharlwood** [comment]
Yeah, so, just been having a play. If I start up the program and the bot, the timer is switched to Production (default) with no timer, it wont work in this stage.

At this point if you hit function - refresh, it then works fine

**Comment by Th3C0D3R** [comment]
It does not work well.. soo I will recode this again to something I hope is the final solution. But unfortunately I can start this at monday

**Comment by andrewcharlwood** [comment]
To be honest it's working fine for me as long as you rerfresh after starting!

also, not really an "issue" per say but just thought i'd mention it. When changing language after logging in I need to manually refresh for it to make changes.

Also it auto-defaults to German every boot, if there was a way of stopping the language being switched on boot that would be handy.

**Comment by Th3C0D3R** [comment]
I will fix that while working on the bot. Making english fist and german as fallback
